release_notes_generator/model/record.py

```diff
@@ -22,18 +22,17 @@
 from github.Repository import Repository
 from github.Commit import Commit
 
-from release_notes_generator.utils.constants import Constants
+from release_notes_generator.utils.constants import (PR_STATE_CLOSED, ISSUE_STATE_CLOSED, ISSUE_STATE_OPEN,
+    RELEASE_NOTE_DETECTION_PATTERN, RELEASE_NOTE_LINE_MARK)
 from release_notes_generator.utils.pull_reuqest_utils import extract_issue_numbers_from_body
 
 
+# pylint: disable=too-many-instance-attributes, too-many-public-methods
```

Comment:
I'm not sure how, but I would suggest splitting this large class into several smaller ones, each focused on a single responsibility.

Reply:
I spent some time thinking and trying to figure out how to do a split. This class represents a universal Record that can behave like an isolated Issue, or PR, or as a rich Issue with several PR.
This class hides all these details and provides the same API for the code.

Options:
- Create sub-record classes for Issue and PullRequest. It leads to moving the logic to manage differences outside of the Record class
- Create Record as abstract parent class defining required method interface. Child classes can provide their own logic implementation. This again requires generator logic to recognize what type of Record it is.

Why to do it?
- The class is too big - Yes.
- There are many property methods. - Yes, but it represents the logic API. So the count will not decrease.
- There are many public methods. - Yes, but it provides the information for building record text representation.

I will make TODO here to recheck it before Release.